eng/tools/lint-diff/src/lint-diff.ts

```diff
@@ -0,0 +1,303 @@
+import { readFile } from "fs/promises";
```

Comment:
TODO: 

* Write a `README.md` file that describes how to run the tool locally
* Tests
    * e2e (consider memfs for mocking filesystem) 
    * unit tests